src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/LocalAppContextSwitches.cs

```diff
@@ -15,8 +15,8 @@ internal static partial class LocalAppContextSwitches
 
         private static bool? s_legacyRowVersionNullBehavior;
         private static bool? s_suppressInsecureTLSWarning;
-        private static bool s_makeReadAsyncBlocking;
-        private static bool s_useMinimumLoginTimeout;
+        private static bool? s_makeReadAsyncBlocking;
```

Comment:
bool? is a structure with two fields. It is not updated atomically. The lock-free reads and writes below have race conditions.

If you want the reads to be lock-free, you can either:
- Change the type of these caches to `bool` and initialize them eagerly in a static constructors
or
- Change the type of these caches to atomically updated integer with 3 possible values uninitialized/false/true. Here is an example of similar pattern in dotnet/runtime: https://github.com/dotnet/runtime/blob/b88c74a109903de213b049177cbc0481a689477f/src/libraries/System.Private.CoreLib/src/System/Globalization/TextInfo.cs#L22C22-L27

Reply:
Thanks. I've updated all the variables to use the tristate enum pattern.